### PR TITLE
Improve integration tests

### PR DESCRIPTION
**Pull request by csala**
Fix a problem on the `HyperTransformer` type detection when being passed a `bool` column with `nan` values, which was incorrectly being labeled as `categorical`.

Also re-organize and clean up a bit the HyperTransformer integration tests.

### CODE REVIEW

File: tests/integration/test_hyper_transformer.py [hunk 1]
```diff
@@ -238,7 +183,7 @@ def test_hypertransformer_field_transformers():
     till it is.
 
     Setup:
-        - The datetime column is set to use a dumm transformer that stringifies
+        - The datetime column is set to use a dummy transformer that stringifies
```
Comment (amontanez24) [inline tests/integration/test_hyper_transformer.py:186]: good catch!

File: tests/integration/test_hyper_transformer.py [hunk 2]
```diff
@@ -332,99 +280,87 @@ def test_hypertransformer_field_transformers_multi_column_fields():
...
 
+    # assert
     pd.testing.assert_frame_equal(subset, reverse)
 
 
 def test_with_unfitted_columns():
     """HyperTransform should be able to transform even if there are unseen columns in data."""
-    data = get_input_data_without_nan()
-
+    # setup
```
Comment (amontanez24) [inline tests/integration/test_hyper_transformer.py:349]: minor: I noticed some files capitalize these. I think we should try and keep consistent, so either have it capitalized everywhere or lower case everywhere

File: tests/integration/test_hyper_transformer.py [hunk 3]
```diff
@@ -83,110 +85,60 @@ def _reverse_transform(self, data):
...
-            'class': 'BooleanTransformer'
-        },
-        'datetime': {
-            'class': 'DatetimeTransformer'
-        },
-        'names': {
-            'class': 'CategoricalTransformer',
-        },
-    }
+DETERMINISTIC_CATEGORICAL = deepcopy(DEFAULT_TRANSFORMERS)
```
Comment (amontanez24) [inline tests/integration/test_hyper_transformer.py:137]: minor: could we make this more explicit. Perhaps `DEFAULTS_WITH_DETERMINISTIC_CATEGORICAL`

File: tests/integration/test_hyper_transformer.py [hunk 4]
```diff
@@ -83,110 +85,60 @@ def _reverse_transform(self, data):
...
-    })
-    if index:
-        transformed.index = index
-
-    return transformed
-
-
-def get_transformed_nan_data():
+def get_transformed_data():
+    index = [4, 6, 3, 8, 'a', 1.0, 2.0, 3.0]
```
Comment (amontanez24) [inline tests/integration/test_hyper_transformer.py:113]: Since this is the same in both the input data and transformed, should we pull it out?

**Review by amontanez24 — APPROVED** [review]
A few minor comments but looks good to me!

**Review by fealho — APPROVED** [review]
Looks good 👍

**Pull request merged**